packages/nuxt/src/app/composables/preload.ts

```diff
@@ -34,11 +34,13 @@ function _loadAsyncComponent (component: Component) {
   }
 }
 
-export async function preloadRouteComponents (to: string, router: Router & { _routePreloaded?: Set<string>; _preloadPromises?: Array<Promise<any>> } = useRouter()): Promise<void> {
+export async function preloadRouteComponents (to: RouteLocationRaw, router: Router & { _routePreloaded?: Set<string>; _preloadPromises?: Array<Promise<any>> } = useRouter()): Promise<void> {
   if (process.server) { return }
 
+  const { path, matched } = router.resolve(to)
+
   if (!router._routePreloaded) { router._routePreloaded = new Set() }
-  if (router._routePreloaded.has(to)) { return }
+  if (router._routePreloaded.has(path)) { return }
```

Comment:
maybe we can also early-return if there are no matched components?

Reply:
Like this?

```ts
if (!matched.length) { return }
if (!router._routePreloaded) { router._routePreloaded = new Set() }
if (router._routePreloaded.has(path)) { return }
```

I think it looks better this way, but I'm not sure when there is no matching component?

Reply:
Looks perfect!